### DISCUSSION TITLE
[3] UCRS and Activity Diagrams

### DISCUSSION THREAD

**Discussion by GermanJesus-lul** [GermanJesus-lul/Rezeptbuch#4] [Announcements]
Hey everyone!

This week we added activity diagrams to three of our Use Cases:
- [Show Recipe](https://github.com/GermanJesus-lul/Rezeptbuch/blob/main/docs/UseCases/UC1_ShowRecipe.md)
- [Navigate Through Recipe](https://github.com/GermanJesus-lul/Rezeptbuch/blob/main/docs/UseCases/UC2_NavigateThroughRecipe.md)
- [List Online Recipes](https://github.com/GermanJesus-lul/Rezeptbuch/blob/main/docs/UseCases/UC4_ListOnlineRecipes.md)

We also created two Use-Case-Requirements-Specifications:
- [Show Recipe](https://github.com/GermanJesus-lul/Rezeptbuch/blob/main/docs/UCRS/UCRS_ShowRecipe.md)
- [Navigate Through Recipe](https://github.com/GermanJesus-lul/Rezeptbuch/blob/main/docs/UCRS/UCRS_NavigateThroughRecipe.md)

Next week, we'll plan the recipe-saving process and tackle upcoming tasks.

Yours truly,
Rezeptbuch Team

**Comment by Unleqit** [comment]
Hey,

your documentation looks well-structured and its contents are easy to understand. Great work!

I have a few questions regarding your UCRS for the "Show Recipe" Use Case: 
First, how are client/server-side errors handled? (e.g. no internet connection, database unreachable, ...)
And second, does each recipe have to be unique or should it possible for two distinct users to upload the same recipe? If this is unintended, is there a check performed at some point before a user is able to publish their recipe?

Regarding your diagrams, I think they are generally in good shape, but it would be beneficial to separate the "Application" node into two "Frontend" and "Backend" nodes in order to visualize some processes in greater detail (e.g. what is shown to the user of your .NET MAUI application when an error occurs somewhere?)

Best regards
> Restaurant Reservierung

**Reply by GermanJesus-lul** [reply]
Hey,
Thank you for your reply.
In the "Show Recipe" Use Case we only use the server to check if the recipe can be published. And as our focus is on offline use there does not need to be much internet connection error handling when first showing the recipe. We will ad some feedback in form of a "greyed-out" publish button and maybe a small tooltip in form of a "no internet connection"-symbol.
When a user actually clicks the publish button there will of course be more feedback to errors. In which form is not described yet.

We are still discussing how we will save and manage our data, but as we want to stay away from a user-based design and want to allow multiple similar recipes (e.g. two slightly different pancake recipes) we are not sure yet how we will handle this. Especially because we also want to allow editing of recipes.

We will look into your point about the diagrams.

Yours truly,
Julius and the Rezeptbuch Team
